docusaurus/docs/migration-guides/update-from-grafana-versions/v11.6.x-v12.0.x.md

```diff
@@ -154,3 +154,100 @@ The following table summarizes the API changes with notes explaining the key dif
 | `PluginExtensionLinkConfig`               | `PluginExtensionAddedLinkConfig`                                                       | Updated type definition for link configuration                                    |
 | `GetPluginExtensionsOptions`              | `UsePluginLinksOptions` or `UsePluginComponentsOptions` or `UsePluginFunctionsOptions` | Updated type definition for hook parameters                                       |
+
+## Migrate from `Select` to `Combobox`
+
+Grafana 11.5.0 introduced the new `Combobox` component as a replacement for the `Select` component. The `Combobox` component has been designed with performance in mind, particularly for handling extremely large data sets. See the [component documentation](https://developers.grafana.com/ui/latest/index.html?path=/docs/forms-combobox--docs) for detailed usage guidelines.
+
+The `Select` and `MultiSelect` components are now deprecated in Grafana 12 and will not receive substantial updates or fixes. We recommend migrating to `Combobox` and `MultiCombobox`.
+
+### Basic usage
+
+For simple use cases, the `Combobox` component has a similar API to `Select`.
+
+```tsx
+/**
+ * Previous Select implementation
+ */
+import { SelectableValue } from '@grafana/data';
+import { Select } from '@grafana/ui';
+
+const options: SelectableValue[] = [
+  { label: 'Loki', value: 'datasource-loki' },
+  { label: 'Prometheus', value: 'datasource-prom' },
+];
+const [value, setValue] = useState<SelectableValue>();
+
+<Select value={value} options={options} onChange={(newOption) => setValue(newOption)} />;
+
+/**
+ * New Combobox implementation
+ */
+import { Combobox, ComboboxOption } from '@grafana/ui';
+
+const options: ComboboxOption[] = [
+  { label: 'Loki', value: 'datasource-loki' },
+  { label: 'Prometheus', value: 'datasource-prom' },
+];
+
+// It is preferred to pass in only the 'value' property from options to the value prop
+const [value, setValue] = useState<string>();
+
+<Combobox value={value} options={options} onChange={(newOption) => setValue(newOption.value)} />;
+```
+
+### Async data loading
+
+The `Combobox` component handles asynchronous data loading through a function passed to the `options` prop. This function:
+
+- Receives the current input value as a parameter
+- Returns a Promise resolving to `ComboboxOption[]`
+- Automatically debounces calls to prevent excessive API requests
+
+```tsx
+import { Combobox, ComboboxOption } from '@grafana/ui';
+
+const [value, setValue] = useState<string>();
+
+const loadOptions = useCallback(async (input: string): Promise<ComboboxOption[]> => {
+  const response = await fetch(`/api/options?query=${input}`);
+  return response.json();
+}, []);
+
+<Combobox value={value} options={loadOptions} onChange={(newOption) => setValue(newOption.value)} />;
```

Comment:
If the value and the label differ, it might be worth mentioning that you can set an object.

Reply:
I am okay with that